src/monty/io.py

```diff
@@ -21,33 +21,42 @@
     from typing import IO, Generator, Union
 
 
-def zopen(filename: Union[str, Path], *args, **kwargs) -> IO:
+def zopen(filename: Union[str, Path], /, mode=None, **kwargs) -> IO:
     """
-    This function wraps around the bz2, gzip, lzma, xz and standard Python's open
-    function to deal intelligently with bzipped, gzipped or standard text
-    files.
+    This function wraps around `[bz2/gzip/lzma].open` and `open`
+    to deal intelligently with compressed or uncompressed files.
+    Supports context manager `with zopen`.
+
+    Note:
+        - explicit binary/text in `mode`
+        - encoding # TODO:
 
     Args:
-        filename (str/Path): filename or pathlib.Path.
-        *args: Standard args for Python open(..). E.g., 'r' for read, 'w' for
-            write.
-        **kwargs: Standard kwargs for Python open(..).
+        filename (str | Path): Filename.
+        # TODO: mode use Literal type
+        mode (str):  E.g., "r" for read, "w" for write.  # TODO: finish this
+        **kwargs: Keyword arguments to pass to `open`.
 
     Returns:
-        File-like object. Supports with context.
+        TextIO | BinaryIO
     """
-    if filename is not None and isinstance(filename, Path):
```

Comment:
I would drop the `filename` as `None` check and merge `Path` conversion as I don't think opening `None` is a legal usage